api/src/convert/account_key.rs

```diff
@@ -7,24 +7,13 @@ use mc_account_keys::AccountKey;
 
 impl From<&AccountKey> for external::AccountKey {
     fn from(src: &AccountKey) -> Self {
-        let mut dst = external::AccountKey::new();
-
-        dst.set_view_private_key(external::RistrettoPrivate::from(src.view_private_key()));
-        dst.set_spend_private_key(external::RistrettoPrivate::from(src.spend_private_key()));
-
-        if let Some(url) = src.fog_report_url() {
-            dst.set_fog_report_url(url.to_string());
-        }
-
-        if let Some(spki) = src.fog_authority_spki() {
-            dst.set_fog_authority_spki(spki.to_vec());
+        Self {
+            view_private_key: Some(src.view_private_key().into()),
+            spend_private_key: Some(src.spend_private_key().into()),
+            fog_report_url: src.fog_report_url().unwrap_or_default().into(),
```

Comment:
instead of conditional assignment using `unwrap_or_default()` to achieve the same behavior but in a struct initializer